docs/design/proposal-troubleshoot-consolidated-cli.md

```diff
@@ -0,0 +1,166 @@
+
+### Previous related proposals
+* https://github.com/replicatedhq/troubleshoot/blob/em/original-proposal-troubleshoot-cli/docs/design/proposal-sbctl-integration.md
+* https://github.com/replicatedhq/troubleshoot/blob/em/original-proposal-troubleshoot-cli/docs/design/proposal-consolidate-collection.md
+
+
+## High Level Design
+
+Functions of `support-bundle`, `preflight`, `analyze`, `redact`, and `sbctl` binaries/tools should be rolled together into a single `troubleshoot` CLI that can perform all necessary functions.
+
+`troubleshoot` CLI should be able to report on the version that is installed in the CLI, and any support bundle generated with `troubleshoot` should report its build/version inside the archive it generates.
+
+## Detailed Design
+
+In the interest of being able to work on this quickly without breaking existing use-cases, a new `troubleshoot` command should be created utilizing cobra and viper best practices from the cobra.dev docs.
+
+To enable the new CLI to be written in a clean and DRY way, we should first address the need for a stable public API for troubleshoot.
+
+A guiding principle of the design of the "Public API" should be that the we are defining a set of "artefacts" (i.e: a support bundle, a spec), and each of the defined public functions acts as an interface to interact with these artefacts, either creating, manipulating, or performing other actions on them. In this way we guarantee that each public function can be re-used on an artefact (support bundle) in any stage of it's existance.
+
+For example, running a set of redactors or analyzers on an existing support bundle without having to re-run collection.
+
+### Troubleshoot top-level commands to subcommands
+
+All top level commands (`support-bundle`, `preflight`, `analyze`, `redact`, and `sbctl`) of the Troubleshoot project will now be subcommands e.g `sbctl` is planned to be `troubleshoot inspect`. Some flags and subcommands names may change to best fit completeness of the CLI interface e.g `--interactive=false` will be `--no-input`. Putting the CLI interface together should aim to follow best practises from well known guidelines such as https://clig.dev/ and other mature projects.
+
+- `troubleshoot inspect` - This subcommand will be equivalent to the `sbctl` command.
+- `troubleshoot collect` - This subcommand will be equivalent to the `support-bundle` command.
+- `troubleshoot preflight` - This subcommand will be equivalent to the `preflight` command.
+- `troubleshoot redact` - This subcommand will be equivalent to the `redact` command.
+- `troubleshoot analyze` - This subcommand will be equivalent to the `analyze` command.
+
+### Public APIs
+
+The existing package entrypoint we use from the support-bundle command is `supportbundle.CollectSupportBundleFromSpec()`
+which is an end-to-end function that runs collectors, redactors and then analysis.
```

Comment:
let's be sure to mention that we do not intend to break this entrypoint immediately, but that it is intended that we use this API to allow future versions to make breaking changes to that function.